### DISCUSSION TITLE
#16 📑 Weekly Report 16

### DISCUSSION THREAD

**Discussion by fehlend0** [SE-TINF22B6/Statify#120] [Blog]
Hello everyone!👋 We are glad to share our weekly news with you. 

## Project Retrospective

We ran a project retrospective meeting where every team member provided their thoughts on the work we have been doing over the past two semesters. To make it more illustrative, we used a template on Miro. Here you can have a look at [the result](https://github.com/SE-TINF22B6/Statify/blob/main/Documentation/Quick_Retrospective.pdf).

As you can see, we identified both positive and negative aspects. However, we were still able to learn a lot and improve our hard and soft skills during this project.

## Accomplishments in the Backend

From our previous reviews, you might know that the data we were using in the development process was mocked. This week, we obtained almost all the real data we need. We implemented logic for retrieving user profile information, user playlists, and user top tracks and artists over three different time ranges. This functionality was based on Spotify API calls and extensive parsing.

We also added two more endpoints: the first one to delete data and the second one to depict the features of a single track. It is one of the tasks for next week to implement the logic for retrieving this information from the Spotify API.

Furthermore, we still pay a lot of attention to the quality of the code. Therefore, some parts were either removed or restructured. For example, we removed the login and register pages, which we had implemented earlier, since we want to use Spotify login only.

## New Tests

Since quality assurance is one of our priorities, we implemented some new tests in the frontend to ensure that our action bar, buttons, and toggle buttons function as intended. Further tests are to come next week!

Thanks for your interest in our project over the past two semesters!

Your Statify Team

**Comment by giomskii** [comment]
Good evening Team Statify,
its nice to see that you can now retrieve actual data from the database. Moreover, its also good that you have also put a little focus on refactoring and restructuring the code before it is to late or confusing. 
Keep up the good work until the final demo 👍🏼  - Team DHBWhub

**Comment by MaarcS** [comment]
Dear Statify team,

it's fantastic to see the progress you've made, especially with integrating real data from Spotify and adding new endpoints. It sounds like the retrospective was a valuable exercise for you; I'm sure you can learn from both the positive and negative aspects that you pointed out. It is also nice to hear that you put a lot of effort into code quality, which definitely helps to achieve a well polished project.

We are excited to watch your presentation next week.

Your CookHub Team

**Comment by Elizabethstella** [comment]
Awesome update the Statify team! The backend improvements and new frontend tests are impressive. Excited to see the focus on quality and upcoming features!

BookHub team.
